File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        vector<int> memo(nums.size(), -1);
        return dfs(nums, 0, memo);
    }

    int dfs(vector<int>& nums, int i, vector<int>& memo) {
        if (i >= nums.size()) {
            return 0;
        }
        if (memo[i] != -1) {
            return memo[i];
        }

        int skip = dfs(nums, i + 1, memo);
        int rob = nums[i] + dfs(nums, i + 2, memo);

        memo[i] = max(skip, rob);
        return memo[i];
    }
};

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_ones = count = 0
        for n in nums:
            if n == 0:
                count = 0
            else:
                count += 1
            max_ones = max(max_ones, count)
        return max_ones